.github/workflows/workflow.yaml

```diff
@@ -317,6 +317,60 @@ jobs:
             done
           done
 
+  benchmark:
+    name: "Benchmark"
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+
+      - name: Download release benchmark file
+        uses: robinraju/release-downloader@v1.6
+        id: get-latest-benchmark
+        continue-on-error: true
+        with: 
+          latest: true
+          fileName: "release_benchmarks.txt"
+
+      - name: Run benchmarks
+        run: |
+          curl -L -O "https://github.com/kubernetes-sigs/kubebuilder/releases/download/v${KUBEBUILDER_VERSION}/kubebuilder_${KUBEBUILDER_VERSION}_linux_amd64.tar.gz" &&\
+            tar -zxvf kubebuilder_${KUBEBUILDER_VERSION}_linux_amd64.tar.gz &&\
+            sudo mv kubebuilder_${KUBEBUILDER_VERSION}_linux_amd64 /usr/local/kubebuilder
+          make benchmark-test BENCHMARK_FILE_NAME=pr_benchmarks.txt
+        if: ${{ steps.get-latest-benchmark.outcome == 'success' }}
+        env:
+          KUBEBUILDER_VERSION: 2.3.1
+
+      - name: Compare benchmarks
+        id: get-comment-body
+        if: ${{ steps.get-latest-benchmark.outcome == 'success' }}
+        run: |
+          export GOPATH="$HOME/go"
+          PATH="$GOPATH/bin:$PATH"
+          go install golang.org/x/perf/cmd/benchstat@latest
+          benchstat release_benchmarks.txt pr_benchmarks.txt > benchstat.txt
+          delimiter=$(openssl rand -hex 8)
+          echo 'msg<<$delimiter' >> $GITHUB_OUTPUT
+          cat benchstat.txt >> $GITHUB_OUTPUT
+          echo '$delimiter' >> $GITHUB_OUTPUT
+        
+      - name: Create commit comment
+        if: ${{ steps.get-latest-benchmark.outcome == 'success' }}
+        uses: peter-evans/create-or-update-comment@v2
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body: |
+            This PR compares its performance to the latest released version. If it performs significantly lower, consider optimizing your changes to improve the performance.
```

Comment:
can we define what significantly lower means? e.g. `5% ns/op delta, etc`

Reply:
And on further thought, could we actually enhance this action to some processing of the results? Like show which benches have changed meaningfully?

As I read this right now, do we comment out the whole benchmark output? That may be a bit too much

Reply:
Let me look into this and the capabilities of `benchstat` to see if this is possible or not.

Reply:
Good point on defining delta. Never gave that a thought! Good catch

Reply:
What do you think the delta should be? Do you know any tools that could process the benchstat output to prettify it? As of now, we are posting the complete result of benchstat on PR.

Reply:
We can pick a value to start with and then adjust as needed. But if we are only running this job thru the comment cmd `/benchmark`, then maybe we don't need to set a threshold and just rely on one off analyses from the pr authors & reviewers